elastic/beats
# [Elastic Agent] Enable log shipping of endpoint-security by Elastic Agent

## What does this PR do?



Starts shipping the logs of endpoint-security to elasticsearch.

This also refactors the code some to pass the `program.Spec` around instead of needing to keep checking for it from the SupportedMap. This is how the code path determines the path for the endpoint-security log paths.

## Why is it important?



So all log information from an Elastic Agent running endpoint-security can be observed.

## Checklist




## How to test this PR locally



Build the Elastic Agent with endpoint-security manually placed in the download folder. Enable Endpoint Security in Fleet and notice that all the Endpoint Security logs show up with `event.dataset: elastic_agent.endpoint-security`

## Related issues


- Closes #21796
